frontend/src/app/modules/auth/sign-in/sign-in.component.ts

```diff
@@ -14,35 +14,46 @@ import { validationErrorMessage } from '../sign-up/error-helper';
 })
 export class SignInComponent {
     form: FormGroup = new FormGroup({
-        email: new FormControl('', [Validators.required, Validators.email, Validators.maxLength(30)]),
-        password: new FormControl(
-            '',
-            [
-                Validators.required,
-                Validators.minLength(6),
-                Validators.maxLength(25),
-                Validators.pattern(passFormatRegex),
-            ],
-        ),
+        email: new FormControl('', {
+            validators: [Validators.minLength(3), Validators.maxLength(50), Validators.pattern(emailRegex), Validators.required],
+            updateOn: 'submit',
+        }),
+        password: new FormControl('', {
+            validators: [Validators.pattern(passwordRegex), Validators.required],
+            updateOn: 'submit',
+        }),
     });
 
+    isChanged = false;
+
     constructor(
         private formBuilder: FormBuilder,
         private authService: AuthService,
         private toastr: ToastrService,
         private router: Router,
-    ) {}
+    ) { }
 
     public signIn() {
-        this.authService
-            .signIn(this.email.value, this.password.value)
-            .then(() => {
-                this.toastr.success('Successfully sign in', 'Sign in');
-                this.router.navigate(['timetable']);
-            })
-            .catch((error) => {
-                this.toastr.error(error.message, 'Sign up');
-            });
+        this.isChanged = true;
+        if (this.form.valid) {
+            this.authService
+                .signIn(this.email.value, this.password.value)
+                .then(() => {
+                    this.toastr.success('Successfully sign in', 'Sign in');
+                    this.router.navigate(['timetable']);
+                })
+                .catch((error) => {
+                    let errorMessage: string | undefined;
+
+                    if (error.code === 'auth/user-not-found') {
+                        errorMessage = this.GetErrorMessageByKey('notExists');
+                    }
+                    if (error.code === 'auth/wrong-password') {
+                        errorMessage = this.GetErrorMessageByKey('incorrectPassword');
+                    }
```

Comment:
It's better to move it to a separate function and make switch/case

Reply:
done